components/post_view/post_message_preview/post_message_preview.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {Post, PostPreviewMetadata} from 'mattermost-redux/types/posts';
+import UserProfileComponent from 'components/user_profile';
+import {UserProfile} from 'mattermost-redux/types/users';
+import Avatar from 'components/widgets/users/avatar';
+import * as Utils from 'utils/utils';
+import PostMessageView from 'components/post_view/post_message_view';
+
+import Timestamp from 'components/timestamp';
+import PostAttachmentContainer from '../post_attachment_container/post_attachment_container';
+import {FormattedMessage} from 'react-intl';
+
+export type Props = {
+    user?: UserProfile;
+    post?: Post;
+    metadata: PostPreviewMetadata;
+};
+export default class PostMessagePreview extends React.PureComponent<Props> {
+    render() {
+        const {user, metadata, post} = this.props;
+        if (post && user) {
+            return (
+                <PostAttachmentContainer
+                    className='permalink'
+                    link={`/${metadata.team_name}/pl/${metadata.id}`}
```

Comment:
Actually I just tested this with subpaths, it will work fine because webpack is aware of the siteURL so it will treat whatever subpath you have configured as the root.